controllers/component_build_controller_simple_build.go

```diff
@@ -83,12 +88,32 @@ func (r *ComponentBuildReconciler) SubmitNewBuild(ctx context.Context, component
 	simpleBuildPipelineCreationTimeMetric.Observe(time.Since(component.CreationTimestamp.Time).Seconds())
 
 	log.Info(fmt.Sprintf("Build pipeline %s created for component %s in %s namespace using %s pipeline from %s bundle",
-		buildPipelineRun.Name, component.Name, component.Namespace, pipelineRef.Name, pipelineRef.Bundle),
+		buildPipelineRun.Name, component.Name, component.Namespace, pipelineName, pipelineBundle),
 		l.Action, l.ActionAdd, l.Audit, "true")
 
 	return nil
 }
 
+func getNameAndBundle(pipelineRef *tektonapi.PipelineRef) (string, string, error) {
+	if pipelineRef.Resolver != "" && pipelineRef.Resolver != "bundles" {
+		return "", "", fmt.Errorf("unsupported Tekton resolver %q", pipelineRef.Resolver)
+	}
+
+	name := pipelineRef.Name
+	var bundle string
+
+	for _, param := range pipelineRef.Params {
+		switch param.Name {
+		case "name":
+			name = param.Value.StringVal
+		case "bundle":
```

Comment:
is it possible not to have 'bundle' specified?

if yes, we  should check it and return error if that is the case

Reply:
It is possible. Build-service will fail while resolving the bundle, same as it would fail before if a user created a BuildPipelineSelector with no `bundle`

```yaml
pipelineRef:
  name: docker-build
  # bundle: ...
```

Reply:
Specifically, the error looks like this and is considered transient
```
Pipelines as Code provision transient error {"controller": "component", "controllerGroup": "appstudio.redhat.com", "controllerKind": "Component", "Component": {"name":"devfile-sample-python-basic","namespace":"default"}, "namespace": "default", "name": "devfile-sample-python-basic", "reconcileID": "2b6c543b-5804-4b92-9cae-14464b74f2b4", "error": " is an unparseable image reference: could not parse reference: "}
```
I can add a permanent error for this too

Reply:
yeah in that case I'd return permanent error in that function when bundle is empty